### DISCUSSION TITLE
Not a normal PiGallery2 usage

### DISCUSSION THREAD

**Discussion by timnis** [bpatrik/pigallery2#787] [General]
Hi, I know this is not how PG2 was designed to use but I still ask.

I just testing to use PG2 to view CAD milling/lathe drawings (.pdf) and steel laser cutting drawings (.dxf) which are converted to png images (useing pdftoppm and ezdxf).  PG2 works really fast showing those small images (<200kb). However I have problems when I generate updates of the drawings and copying to PG2's images folder to replace older ones . After copying  PG2 fails to index those again. I have try reset PG2 keeping configs but not succeed.

So, is there a way to replace all or some image (images may not be exact copies) in PG2 and get indexing/thumbnails creation to work? Or maybe delete old  container and create new one with keeping same config?

**Comment by bpatrik** [comment]
I think your issue is that the thumbnails are already created for that file
name so they are not recreated.

A workaround is to rename your files.

**Comment by bpatrik** [comment]
Alternatively delete the temp folder. (or selectively delete files from
there)

**Reply by timnis** [reply]
So it should work if I replaces images in PG2 images folder and delete temp content?

**Comment by bpatrik** [comment]
Short answer yes.

Pigallery2 does not considered modified time when creating the name of the
thumbnail and preview (big thumbnail) photo.

So you weather need to rename the original photo as the name of the
original photo name is used to generate the thumbnail photo name.

If a thumbnail already exists the app won't recreate it.


If you want a better solution, I would appreciate it you could send a PR
that adds last modified time to the thumbnail name here:

https://github.com/bpatrik/pigallery2/blob/master/src/backend/model/fileaccess/fileprocessing/PhotoProcessing.ts#L111-L122

This also has a pair where it checks if the name is valid. I had some
similar commits recently you can check that.



Also:
Pg2 now has some extension support, you should be able to implement an
image renderer for PDFs so pdf would work out of the box. (Zoom won't work
tho as during zoom the app loads the original file)


-- Sorry for being brief, sent from my phone.

**Comment by timnis** [comment]
Just tested with delete tmp folder and copy new images.
now thumbnail generation fails, any ideas?
```
11/27/2023, 2:36:52 PM[INFO_][JOB] Running job : Thumbnail Generation
11/27/2023, 2:36:53 PM[ERROR][JOB] Job failed with:
11/27/2023, 2:36:53 PM[ERROR][JOB] TypeError: Cannot read properties of null (reading 'creationDate')
    at /app/src/backend/model/threading/DiskMangerWorker.js:182:63
    at Array.forEach (<anonymous>)
    at DiskMangerWorker.scanDirectory (/app/src/backend/model/threading/DiskMangerWorker.js:181:29)
    at async DiskManager.scanDirectory (/app/src/backend/model/DiskManger.js:30:26)
    at async ThumbnailGenerationJob.loadADirectoryFromDisk (/app/src/backend/model/jobs/jobs/FileJob.js:111:25)
    at async ThumbnailGenerationJob.step (/app/src/backend/model/jobs/jobs/FileJob.js:67:17)
    at async /app/src/backend/model/jobs/jobs/Job.js:101:22
```

**Comment by bpatrik** [comment]
Cab you enable debug logs?

-- Sorry for being brief, sent from my phone.

**Reply by timnis** [reply]
Hi, with debug no much more

```
[pigallery2] | 11/28/2023, 10:22:45 AM[DEBUG] POST /pgapi/admin/jobs/scheduled/Thumbnail%20Generation/start 200 13ms
[pigallery2] | 11/28/2023, 10:22:45 AM[SILLY][DiskManager] scanning directory: /
[pigallery2] | 11/28/2023, 10:22:45 AM[VERBS] GET /pgapi/admin/jobs/scheduled/progress 200 1ms
[pigallery2] | 11/28/2023, 10:22:45 AM[ERROR][JOB] Job failed with:
[pigallery2] | 11/28/2023, 10:22:45 AM[ERROR][JOB] TypeError: Cannot read properties of null (reading 'creationDate')
[pigallery2] |     at /app/src/backend/model/threading/DiskMangerWorker.js:182:63
[pigallery2] |     at Array.forEach (<anonymous>)
[pigallery2] |     at DiskMangerWorker.scanDirectory (/app/src/backend/model/threading/DiskMangerWorker.js:181:29)
[pigallery2] |     at async DiskManager.scanDirectory (/app/src/backend/model/DiskManger.js:30:26)
[pigallery2] |     at async ThumbnailGenerationJob.loadADirectoryFromDisk (/app/src/backend/model/jobs/jobs/FileJob.js:111:25)
[pigallery2] |     at async ThumbnailGenerationJob.step (/app/src/backend/model/jobs/jobs/FileJob.js:67:17)
[pigallery2] |     at async /app/src/backend/model/jobs/jobs/Job.js:101:22
[pigallery2] | 11/28/2023, 10:22:49 AM[VERBS] GET /pgapi/admin/jobs/scheduled/progress 200 2ms
[pigallery2] | 11/28/2023, 10:22:54 AM[VERBS] GET /pgapi/admin/jobs/scheduled/progress 304 4ms
```

**Reply by timnis** [reply]
file permission are ok because I can read file from inside pg2 container
```
$ podman exec -it pigallery2 /bin/bash
root@aa72a14ccbb1:/app# stat data/images/po_A100141.png 
  File: data/images/po_A100141.png
  Size: 45076     	Blocks: 96         IO Block: 4096   regular file
Device: fd00h/64768d	Inode: 25723341    Links: 1
Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2023-11-28 10:27:09.683924045 +0000
Modify: 2023-11-27 14:31:18.493529847 +0000
Change: 2023-11-28 10:22:28.595910748 +0000
 Birth: 2023-11-27 14:31:18.493529847 +0000
```

**Reply by timnis** [reply]
No idea why it's working now, but it does :)

I running now following 
```
podman stop pigallery2
rm -rf ${pigallery2}/tmp/tc/*.*
rm -rf ${pigallery2}/images/*.png
cp *.png ${pigallery2}/images/
podman start pigallery2 
```

**Reply by bpatrik** [reply]
That looks ok to me. Not sure why you had the issue before.

**Reply by timnis** [reply]
Me either 😄  Thanks for the help!
